lighting/models/product_source_line.py

```diff
@@ -380,6 +440,64 @@ def _compute_wattage_display(self):
         string="Source",
     )
 
+    invisible_fields = fields.Json(
+        compute="_compute_invisible_fields",
+    )
+
+    def _get_all_dependent_fields(self):
+        self.ensure_one()
+        all_fields = set()
+        for fields_g in self.FIELD_GROUP_MAP().values():
+            all_fields |= set(fields_g)
+        return all_fields
+
+    def _get_irrelevant_dependent_fields(self, values=None):
+        self.ensure_one()
+        return self._get_all_dependent_fields() - self._get_relevant_dependent_fields(
+            values=values
+        )
+
+    def _get_relevant_dependent_fields(self, values=None):
+        self.ensure_one()
+        if not values:
+            new_vals = self
+        else:
+            new_vals = {}
+            for x in self.KEY_FIELDS():
+                new_vals[x] = values[x] if x in values else self[x]
+        key = "".join(["1" if new_vals[x] else "0" for x in self.KEY_FIELDS()])
+        group_values = self.GROUP_MAP()[key]
+        field_group_map = self.FIELD_GROUP_MAP()
+        fields = set()
+        for group, relevant in group_values.items():
+            if relevant:
+                fields |= field_group_map[group]
+        return fields
+
+    def get_dependent_field_value(self, field):
+        self.ensure_one()
+        if field not in self._get_all_dependent_fields():
+            raise ValidationError(_("Field %s is not a dependent field" % field))
+        if field in self._get_relevant_dependent_fields():
+            return self[field]
+        return None
+
+    def update_dependent_fields(self, values):
+        self.ensure_one()
+        for field in self._get_irrelevant_dependent_fields(values):
+            # This is to be sure we don't update already falsy values
+            if not self[field]:
+                if field in values:
+                    del values[field]
+            else:
+                values[field] = None
```

Comment:
We should have all these methods at the beginning just after the "constants" and comment the start and end of the block